ext/zlib/zlib.c

```diff
@@ -807,35 +807,29 @@ static bool zlib_create_dictionary_string(HashTable *options, char **dict, size_
 				if (zend_hash_num_elements(dictionary) > 0) {
 					char *dictptr;
 					zval *cur;
-					zend_string **strings = emalloc(sizeof(zend_string *) * zend_hash_num_elements(dictionary));
+					zend_string **strings = safe_emalloc(zend_hash_num_elements(dictionary), sizeof(zend_string *), 0);
 					zend_string **end, **ptr = strings - 1;
 
 					ZEND_HASH_FOREACH_VAL(dictionary, cur) {
-						size_t i;
-
 						*++ptr = zval_get_string(cur);
```

Comment:
In the future we should use the `try` API here.

Reply:
Doesn't make it better, I'm afraid. Now we have to check for `EG(exception)` but can be sure that `*ptr` is not NULL; with `zval_try_get_string()` we no longer have to check for `EG(exception)`, but need to check that `*ptr` is not NULL before releasing it. Basically, what we did.

Reply:
Sure, but it is easier to miss checking for `EG(exception)` and use some sort of default `zend_string` then not having one in the first place IMHO.